scripts/download_backtest_data.py

```diff
@@ -1,50 +1,60 @@
 #!/usr/bin/env python3
 """
-This script generates json data
+This script generates json files with pairs history data
 """
+import arrow
 import json
 import sys
 from pathlib import Path
-import arrow
-from typing import Any, Dict
+from typing import Any, Dict, List
 
-from freqtrade.arguments import Arguments
-from freqtrade.arguments import TimeRange
-from freqtrade.exchange import Exchange
+from freqtrade.arguments import Arguments, TimeRange
+from freqtrade.configuration import Configuration
 from freqtrade.data.history import download_pair_history
-from freqtrade.configuration import Configuration, set_loggers
+from freqtrade.exchange import Exchange
 from freqtrade.misc import deep_merge_dicts
 
 import logging
-logging.basicConfig(
-    level=logging.INFO,
-    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
-)
-set_loggers(0)
+
+logger = logging.getLogger('download_backtest_data')
 
 DEFAULT_DL_PATH = 'user_data/data'
 
 arguments = Arguments(sys.argv[1:], 'download utility')
-arguments.testdata_dl_options()
-args = arguments.parse_args()
+arguments.download_data_options()
+
+# Do not read the default config if config is not specified
+# in the command line options explicitely
+args = arguments.parse_args(no_default_config=True)
 
 timeframes = args.timeframes
+pairs: List = []
 
-if args.config:
-    configuration = Configuration(args)
+configuration = Configuration(args)
+config: Dict[str, Any] = {}
 
-    config: Dict[str, Any] = {}
+if args.config:
     # Now expecting a list of config filenames here, not a string
     for path in args.config:
-        print(f"Using config: {path}...")
+        logger.info(f"Using config: {path}...")
         # Merge config options, overwriting old values
         config = deep_merge_dicts(configuration._load_config_file(path), config)
 
     config['stake_currency'] = ''
     # Ensure we do not use Exchange credentials
+    config['exchange']['dry_run'] = True
     config['exchange']['key'] = ''
     config['exchange']['secret'] = ''
+
+    if args.exchange:
+        config['exchange']['name'] = args.exchange
+
+    pairs = config['exchange']['pair_whitelist']
+    timeframes = [config['ticker_interval']]
```

Comment:
`-t 1d` is ineffective and uses 5m from the configuration if a configuration is used (i guess you could simply check here `if not timeframes: ...`, simillar to the check for `if args.exchange:` two lines above.

(and honestly, -t is one of the best options this script offers as we can download 5m,15m,1d data in one go).

Reply:
Will check this a bit later, busy today...

As a possible further improvement -- if used with the strategy, this script should fetch informative pairs from it and add to the download list along with their timeframes that can differ from timeframes set in the config... What do you think?

Reply:
noone said it needs fixing immediately, so no worries

i wouldn't don't overcomplicate this script tbh - a method to simply fetch all pairs (for a pairlist) and we should be good ... theres a ton of other things to improve ;)

Reply:
Handling of exchange is also broken. If config is used, then the script currently works with bittrex (i.e. default value) even if binance is set in the config. 

All this needs further refactoring... I will probably move handling of the parameters to freqtrade/configuration.py, as for other parameters and options for the main freqtrade. This seems to be most efficient approach and will help make the script a command of main freqtrade in the future...

Reply:
I disagree (and i did test it too), lines 49/50 override the exchange that is set in the configuration with what is set in the parameter.

Reply:
Yes, but if exchange is not given in the command line, it's not taken from the config but a default bittrex is used. That'a because exchange defaults to bittrex in the freqtrade/arguments.py and args.exchange is not empty when it's not specified in the command line...

Reply:
oh sorry, didn't think about that direction ...

Reply:
Something like this, sure -- remove default.
But I'm now thinking that such approach (removal of default in parseargs) should also be implemented for all other options (well, for almost all of them -- the `-v` is probably is a good excemption, it's well working) and assigning of defaults should be moved completely to configuration.py, where they are merged with config... Just because now we have, for example
```
Parameter -s/--spaces detected: all
```
in the log for the hyperopt when no -s option is used...

Of course, for other options it should be done in a sep. PR, so I'm still evaluating all this to find a better common approach...